include/dfu/mcuboot.h

```diff
@@ -17,26 +17,57 @@
 extern "C" {
 #endif
 
+#ifdef BOOT_SWAP_TYPE_NONE
+#if BOOT_SWAP_TYPE_NONE != 1 /*ensure the same definition in MCUboot */
```

Comment:
The only place where `BOOT_SWAP_TYPE_` are needed (but are not provided by MCUBOOT_BOOTUTIL) is mcumgr port. We don't need to do it in this PR, but IMO mcumgr should link to MCUBOOT_BOOTUTIL cmake target to get those macros. After doing so, BOOT_SWAP_TYPE_ macros could be thrown away from this header. This is the last duplicated code from mcuboot I guess.

Reply:
I prefer to do that in another PR.